### PR TITLE
Cluster image selection support

### PR DESCRIPTION
**Pull request by aleksejlopasov**
Now the user can select image version or provide custom image for the cluster.
Initialization-actions support has been added.
Secondary worker config has been added, though nodes are always preemptible.
Variable ALLOW_RANDOM_CLUSTER_NAMES has been added to Jupyterhub config.

### CODE REVIEW

**Review by JerryLeiDing — COMMENTED** [review]
A few minor text cleanups, otherwise this looks good

File: dataprocspawner/spawner.py
Comment (JerryLeiDing) [inline dataprocspawner/spawner.py:1196]: Is this necessary? Dataproc's frontend should do the work of inferring the image version given a custom image URI, there's no reason for us to validate it here as well.

**Review by JerryLeiDing — COMMENTED** [review]
General note: it would be good to expand on the existing form unit test to ensure that all the options are being set as expected. The test can be found in tests/test_spawner.py on line 539.

File: dataprocspawner/spawner.py
Comment (aleksejlopasov) [reply dataprocspawner/spawner.py:1196]: setting up the image version here is important if admins define different softwareConfig.imageVersion in YAML file.
Example: if 1.5-debian10 is set in YAML and custom image (1.5.17-debian10) is selected in UI, you will get an exception:
```
Error: HTTP 500: Internal Server Error (Error in Authenticator.pre_spawn_start:
InvalidArgument 400 Specified image version '1.5-debian10' doesn't match implied version '1.5.17-debian10' from uri 'https://www.googleapis.com/compute/v1/projects/<project-id>/global/images/<image-id>'!)
```
Additionally we check image_version for Component Gateway compatibility, so it has to be set up properly before the validation occurs.

File: dataprocspawner/customize_cluster.py
Comment (aleksejlopasov) [reply dataprocspawner/customize_cluster.py:195]: The new naming convention will be applied in next PR with Material UI

File: dataprocspawner/customize_cluster.py
Comment (aleksejlopasov) [reply dataprocspawner/customize_cluster.py:184]: The new naming convention will be applied in next PR with Material UI

File: dataprocspawner/customize_cluster.py
Comment (aleksejlopasov) [reply dataprocspawner/customize_cluster.py:204]: The new naming convention will be applied in next PR with Material UI

File: dataprocspawner/customize_cluster.py
Comment (aleksejlopasov) [reply dataprocspawner/customize_cluster.py:220]: The new naming convention will be applied in next PR with Material UI

File: dataprocspawner/customize_cluster.py
Comment (aleksejlopasov) [reply dataprocspawner/customize_cluster.py:229]: The new naming convention will be applied in next PR with Material UI

File: dataprocspawner/spawner.py
Comment (JerryLeiDing) [reply dataprocspawner/spawner.py:1196]: We can unset the image version if a custom image is set and vice versa.

The point about image version checking for Component Gateway is well taken. I'm a little concerned about latency implications for a call to GCE before spawning, but let's merge this and see if it actually has an impact.

**Review by JerryLeiDing — APPROVED** [review]

File: dataprocspawner/spawner.py
Comment (JerryLeiDing) [inline dataprocspawner/spawner.py:1121]: Oops, just realized goog_version needs to be initialized to empty string before the try block or else it'll blow up if an exception occurs.

Comment (aleksejlopasov) [reply dataprocspawner/spawner.py:1122]: sorry, my loss. fixed

**Pull request merged**
